gatsby-node.js

```diff
@@ -1,23 +1,98 @@
-// eslint-disable-next-line @typescript-eslint/no-var-requires
 const path = require(`path`);
+const fetch = require('node-fetch');
+const xmlParser = require('fast-xml-parser');
 
-const BLOG_POST_TEMPLATE_PATH = path.resolve(`src/templates/blog-post.tsx`);
-const CLIENT_TEMPLATE_PATH = path.resolve(`src/templates/client-details.tsx`);
+const BLOG_POST_TEMPLATE_PATH = path.resolve('src/templates/blog-post.tsx');
+const CLIENT_TEMPLATE_PATH = path.resolve('src/templates/client-details.tsx');
+const CAREER_TEMPLATE_PATH = path.resolve('src/templates/career.tsx');
+const CAREER_DETAILS_TEMPLATE_PATH = path.resolve(
+  'src/templates/career-details.tsx',
+);
 
-exports.createPages = async ({ actions, graphql, reporter }) => {
+const PERSONIO_JOBS_URL = 'https://satellytes.jobs.personio.de/xml';
+const PERSONIO_SHORT_DESCRIPTION_NAME = 'Kurzbeschreibung';
+
+exports.createPages = async (createPagesArgs) => {
+  await createCareerPages(createPagesArgs);
+  await createBlogPages(createPagesArgs);
+  // await createClientPages(createPagesArgs);
+};
+
+const createCareerPages = async ({ actions }) => {
   const { createPage } = actions;
 
-  const clientPages = await graphql(`
-    {
-      allClientsJson {
-        nodes {
-          path
-        }
-      }
+  const jobsXmlResponse = await fetch(PERSONIO_JOBS_URL);
+  const jobsXml = await jobsXmlResponse.text();
+  const jobsParse = xmlParser.parse(jobsXml);
+  const positions = jobsParse['workzag-jobs'].position;
+
+  const slugifyPositionName = (name) => {
+    return name
+      .toLowerCase()
+      .replace(/ /g, '-')
+      .replace(/-/g, '-')
+      .replace(/\//g, '-')
+      .replace(/\(/g, '')
+      .replace(/\)/g, '');
+  };
+
+  positions.forEach((position) => {
+    position.satellytesPath = `/career/${position.id}-${slugifyPositionName(
+      position.name,
+    )}`;
+
+    // we need to normalize the jobDescription to an array, otherwise gatsby
+    // complains about different types for the same variable
+    position.jobDescriptions.jobDescription = Array.isArray(
+      position.jobDescriptions.jobDescription,
+    )
+      ? position.jobDescriptions.jobDescription
+      : [position.jobDescriptions.jobDescription];
+
+    position.jobDescriptions.jobDescription = position.jobDescriptions.jobDescription.map(
+      (description) => {
+        return {
+          name: description.name.trim(),
+          value: description.value.trim(),
+        };
+      },
+    );
+
+    const description = position.jobDescriptions.jobDescription.find(
```

Comment:
The xml library is fully based on transforming XML into JS data structure. So it doesn't support `xpath` which would be  something like this`/position/jobDescriptions/jobDescription[name=PERSONIO_SHORT_DESCRIPTION_NAME]` which I really like to read. On the other hand just using the everyday tools from JS is nice here too 🙌